app/kuma-dp/cmd/wait.go

```diff
@@ -0,0 +1,80 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+var waitLog = dataplaneLog.WithName("wait")
+
+func newWaitCmd() *cobra.Command {
+	args := struct {
+		url            string
+		requestTimeout time.Duration
+		timeout        time.Duration
+		checkFrequency time.Duration
+	}{
+		url:            "http://localhost:9901/ready",
```

Comment:
Maybe instead of envoy ready we should check listener that is generated by dp?

Reply:
Or maybe we should add some additional function that checks envoy's listener and if there is specific one/or more than one

Reply:
the admin listener with stats is generated at pod IP so I presume this might be a problem. What we can do however is check for listeners in stats to see if we received things already.

Reply:
Ok so I run multiple tests and it seems that /ready returns 200 after we've got all listeners. I first added this
```
req, err = http.NewRequest(http.MethodGet, url+"/stats?filter=listener_manager.lds.version_text", nil)
if err != nil {
	return err
}
resp, err = client.Do(req)
if err != nil {
	return err
}
defer resp.Body.Close()
bytes, err := io.ReadAll(resp.Body)
if err != nil {
	return err
}
if resp.StatusCode != 200 {
	return fmt.Errorf("HTTP status code %v", resp.StatusCode)
}
if string(bytes) == "listener_manager.lds.version_text: \"\"\n" {
	return errors.New("envoy not configured")
}
return nil
```
but it seems pointless. I'd go with just `/ready` and we can see if the test will be flaky. If so we can improve the check further